### PR TITLE
test(ui): Set environment on master branch + CI

### PR DESCRIPTION
**Pull request by scttcper**
CI data can be messy since it includes everyone's PR's that may or may not be up to date. Filtering by branch samples the data.

This would reduce sampling for the master branch by setting the environment to production and reducing the sample rate for the dev environment.

### CODE REVIEW

File: jest.config.ts [hunk 1]
```diff
@@ -276,8 +278,9 @@ const config: Config.InitialOptions = {
       init: {
         // jest project under Sentry organization (dev productivity team)
         dsn: 'https://3fe1dce93e3a4267979ebad67f3de327@sentry.io/4857230',
-        environment: CI ? 'ci' : 'local',
-        tracesSampleRate: 1,
+        // Use production env to reduce sampling of commits on master
+        environment: CI ? (IS_MASTER_BRANCH ? 'production' : 'ci') : 'local',
```
Comment (mattgauntseo-sentry) [inline jest.config.ts:282]: Nit: Could we make this `? 'ci:master' : 'ci:pull_request'` or something along those lines?

`production` just feels a little out of place with what it's trying to portray.

**Review by mattgauntseo-sentry — APPROVED** [review]
Overall love this change as making sense of the data in the javascript project is tough without this.

I've been using the branch tag, but it seems like this isn't going to work for alerting in the near future.

File: jest.config.ts [hunk 2]
```diff
@@ -276,8 +278,9 @@ const config: Config.InitialOptions = {
       init: {
         // jest project under Sentry organization (dev productivity team)
         dsn: 'https://3fe1dce93e3a4267979ebad67f3de327@sentry.io/4857230',
-        environment: CI ? 'ci' : 'local',
-        tracesSampleRate: 1,
+        // Use production env to reduce sampling of commits on master
+        environment: CI ? (IS_MASTER_BRANCH ? 'ci:master' : 'ci:pull_request') : 'local',
+        tracesSampleRate: CI ? 1 : 0.5,
```
Comment (billyvg) [inline jest.config.ts:283]: Do we want to change sample rate by `CI` or `IS_MASTER_BRANCH`?

Comment (scttcper) [reply jest.config.ts:283]: attempting to prioritize CI for now by sending fewer in dev/local. but i might do that too

**Pull request merged**